### PR TITLE
gc_worker: Split config and GCManager out to make the file smaller

### PR DESCRIPTION
**Pull request by MyonKeminta**
Signed-off-by: MyonKeminta <MyonKeminta@users.noreply.github.com>

###  What have you changed?

The GCWorker file is too large. This PR splits config and GCManager to new files to make gc_worker/lib.rs smaller.
Then the remaining of gc_worker/lib.rs is moved to a new gc_worker.rs so that the logs will contains the filename `gc_worker.rs` rather than `lib.rs`, which may cause confusion when reading logs.

###  What is the type of the changes?

- Engineering (engineering change which doesn't change any feature or fix any issue)

###  How is the PR tested?

- Unit test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

NO

###  Does this PR affect `tidb-ansible`?

NO

### CODE REVIEW

File: src/server/gc_worker/mod.rs [hunk 1]
```diff
@@ -137,7 +77,7 @@ impl<T: PdClient + 'static> GcSafePointProvider for Arc<T> {
     }
 }
 
-enum GcTask {
+pub(self) enum GcTask {
```
Comment (sticnarf) [inline src/server/gc_worker/mod.rs]: I've never seen this... What's its purpose?

Comment (MyonKeminta) [reply src/server/gc_worker/mod.rs]: Trying to export to other submodules but do not to export externally. This is removed in the next commit. PTAL again.

Comment (sticnarf) [reply src/server/gc_worker/mod.rs]: It sounds weird. I think it should be naturally visible by submodules...

**Comment by MyonKeminta** [comment]
@youjiali1995 PTAL thanks!

**Comment by youjiali1995** [comment]
```

[2020-02-05T12:36:49.521Z] error[E0432]: unresolved import `tikv::server::gc_worker::GC_MAX_EXECUTING_TASKS`

[2020-02-05T12:36:49.521Z]  --> tests/failpoints/cases/test_gc_worker.rs:7:41

[2020-02-05T12:36:49.521Z]   |

[2020-02-05T12:36:49.521Z] 7 | use tikv::server::gc_worker::{GcWorker, GC_MAX_EXECUTING_TASKS};

[2020-02-05T12:36:49.521Z]   |                                         ^^^^^^^^^^^^^^^^^^^^^^ no `GC_MAX_EXECUTING_TASKS` in `server::gc_worker`
```
Compile failed @MyonKeminta

**Comment by MyonKeminta** [comment]
Oh, sorry

**Comment by siddontang** [comment]
how about the compiling time?

**Comment by youjiali1995** [comment]
> how about the compiling time?

I think it makes no difference. It's just a refactor.

**Pull request merged**